src/utils/getNameOrValue.js

```diff
@@ -25,6 +25,8 @@ export default function getNameOrValue(path: NodePath, raw?: boolean): string {
       return node.name;
     case types.Literal.name:
       return raw ? node.raw : node.value;
+    case types.FunctionTypeParam.name:
```

Comment:
Doing this here seems weird to me. I wouldn't expect `getNameOrValue` to be called with with a type annotation (but who knows...). It's also not quite right, since some `FunctionTypeParam` do have a name.

Personally I feel it makes more sense to fix this in `getFlowType`. If the name of the `FunctionTypeParam` is `null`, assign an empty string instead. Something like

    name: param.node.name ? param.node.name.name || '';

@danez , why does [this](https://github.com/reactjs/react-docgen/blob/67fcd097705eff40dac17c384353d64c3add8daa/src/utils/getFlowType.js#L151) use `getProperty` to extract the name at the moment? Shouldn't it directly by `getNameOrValue(param.get('name'))` or just `param.node.name.name`? 
Am I looking at the wrong syntax tree or am I missing a case here?

Reply:
Good question, I have no idea :) But seems unnecessary as it can never be computed or be a literal. 

Doing `name: param.node.name ? param.node.name.name || '';` should be correct in `getFlowType` unless the tests think different.